src/transformers/modeling_utils.py

```diff
@@ -413,7 +413,7 @@ def tie_weights(self):
         the weights instead.
         """
         output_embeddings = self.get_output_embeddings()
-        if output_embeddings is not None:
+        if output_embeddings is not None and self.config.tie_word_embeddings:
```

Comment:
@stas00 - I added the flag at this point so that `_tie_or_clone_weights` is only called if `self.config.tie_word_embeddings` is `True`. => Thus Reformer will not call this function.